src/coreclr/nativeaot/Runtime/windows/CoffNativeCodeManager.cpp

```diff
@@ -554,11 +559,41 @@ bool CoffNativeCodeManager::UnwindStackFrame(MethodInfo *    pMethodInfo,
     pRegisterSet->pIP = PTR_PCODE(pRegisterSet->SP - sizeof(TADDR));
 
     memcpy(pRegisterSet->Xmm, &context.Xmm6, sizeof(pRegisterSet->Xmm));
-#endif // TARGET_X86
+#elif defined(TARGET_ARM64)
+    for (int i = 8; i < 16; i++)
+    {
+        context.V[i].Low = pRegisterSet->D[i - 8];
+        context.V[i].High = 0;
+    }
+
+    context.Sp = pRegisterSet->SP;
+    context.Pc = pRegisterSet->IP;
+
+    SIZE_T  EstablisherFrame;
+    PVOID   HandlerData;
+
+    RtlVirtualUnwind(NULL,
+                    dac_cast<TADDR>(m_moduleBase),
+                    pRegisterSet->IP,
+                    (PRUNTIME_FUNCTION)pNativeMethodInfo->runtimeFunction,
+                    &context,
+                    &HandlerData,
+                    &EstablisherFrame,
+                    &contextPointers);
+
+    pRegisterSet->SP = context.Sp;
+    pRegisterSet->IP = context.Pc;
+
+    pRegisterSet->pIP = PTR_PCODE(pRegisterSet->SP - sizeof(TADDR));
```

Comment:
How do we know that `pIP` is `SP - sizeof(TADDR)`? I do not think that this is a valid assumption on ARM64.

Reply:
Good catch, I guess it should be `pRegisterSet->pIP = pRegisterSet->pLR`.